### PR TITLE
add feature: notes to applications

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

File: app/client/templates/admin/submissionReview.js [hunk 1]
```diff
@@ -24,19 +16,55 @@ Template.submissionReview.helpers({
...
+    });
+
+    if (note) {
+      return note.note;
+    } 
+    return false;
+  },
+
+  noteByUser: function(userId) {
+    userId = typeof userId !== 'undefined'? userId : Meteor.userId();
```
Comment (varffs) [inline app/client/templates/admin/submissionReview.js:36]: @mixcas where did this come from?

File: app/server/publications.js [hunk 2]
```diff
@@ -57,4 +57,23 @@ Meteor.publish('allRatings', function() {
...
-});
\ No newline at end of file
+});
+
+// Notes
+Meteor.publish('notes', function(userId, applicationId) {
+  check(userId, String);
+
+  if (!Roles.userIsInRole(this.userId, ['admin', 'committee',])) {
+    throw new Meteor.Error('not-allowed', 'You must be more powerful aka No Juice Error', '334');
```
Comment (varffs) [inline app/server/publications.js:68]: @mixcas is this error code unique or does it come from another place?

File: app/server/publications.js [hunk 3]
```diff
@@ -57,4 +57,23 @@ Meteor.publish('allRatings', function() {
...
+  }
+
+  return Notes.find({userId: userId, applicationId: applicationId});
+});
+
+Meteor.publish('allNotes', function() {
+  if (!Roles.userIsInRole(this.userId, ['admin', 'committee',])) {
```
Comment (varffs) [inline app/server/publications.js]: shouldnt this be only admin roles allowed?

File: app/server/methods/notesMethods.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    if (!this.userId) {
+      throw new Meteor.Error('error-not-signed-in', 'You must sign in first.', '654');
+    }
+
+    if (!Roles.userIsInRole(this.userId, ['committee',])) {
+      throw new Meteor.Error('not-allowed', 'You must be a committee user aka No Juice Error');
+    }
+
+    // Update note
+    Notes.update({userId: Meteor.userId(), applicationId: applicationId,}, { $set: { note: noteText } }, { upsert: true});
```
Comment (varffs) [inline app/server/methods/notesMethods.js]: This matches existing notes and replaces data even though no Note ID is supplied? And this creates new notes if not created?

Comment (mixcas) [inline app/server/methods/notesMethods.js]: yeah, it matching thru `userId` and `applicationId`.
Upsert is magick. if it exist, update; else, insert.

File: app/server/publications.js (hunk 3 above)
Comment (mixcas) [inline app/server/publications.js]: True

File: app/server/publications.js (hunk 2 above)
Comment (mixcas) [inline app/server/publications.js:68]: Fixing it

**Comment by mixcas** [comment]
i managed (i actually dont know how it happened) to merge this but without 2122fa7 and 73a81c1
since both commits are not relevant im gonna close this

**Pull request closed without merging**